04/GeoffRiley/usertweets.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+from collections import namedtuple
+import csv
+import os
+
+import tweepy
+
+from config import CONSUMER_KEY, CONSUMER_SECRET
+from config import ACCESS_TOKEN, ACCESS_SECRET
+
+DEST_DIR = 'data'
+EXT = 'csv'
+NUM_TWEETS = 100
+
+Tweet = namedtuple('Tweet', 'id_str created_at text')
+
+
+class UserTweets(object):
+    """TODOs:
+    - create a tweepy api interface
+    - get all tweets for passed in handle
+    - optionally get up until 'max_id' tweet id
+    - save tweets to csv file in data/ subdirectory
+    - implement len() an getitem() magic (dunder) methods"""
+
+    def __init__(self, handle, max_id=None):
+        self._handle = handle
+        self._save_file = f'{DEST_DIR}/{handle}.{EXT}'
+        self._max_id = max_id
+        self._auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
+        self._auth.set_access_token(ACCESS_TOKEN, ACCESS_SECRET)
+        self._api = tweepy.API(self._auth)
+        self.read_tweets()
+        self.save()
+
+    def read_tweets(self):
+        twitterings = self._api.user_timeline(screen_name=self._handle, max_id=self._max_id, count=NUM_TWEETS)
```

Comment:
E501? not sure but I usually would use multiple lines

Reply:
Actually, I don't know why I let such a long line get through… I usually go through and deal with them before I push to git.  I use PyCharm so it commonly flags up if I've missed any too,.  Odd.

I totally agree with you though.